app/src/test/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreenViewModelTest.kt

```diff
@@ -388,48 +271,97 @@ class OtherUserProfileScreenViewModelTest {
     fun `given a userId, when blocking user fails, then show error message and dismiss BlockDialog`() =
         runTest {
             // given
-            coEvery { blockUser(any()) } returns BlockUserResult.Failure(Unknown(RuntimeException("some error")))
-            otherUserProfileScreenViewModel.infoMessage.test {
+            val (arrangement, viewModel) = OtherUserProfileViewModelArrangement()
+                .withBlockUserResult(BlockUserResult.Failure(Unknown(RuntimeException("some error"))))
+                .arrange()
+            viewModel.infoMessage.test {
 
                 // when
                 expectNoEvents()
-                otherUserProfileScreenViewModel.onBlockUser(USER_ID, "some name")
+                viewModel.onBlockUser(USER_ID, "some name")
 
                 // then
-                coVerify {
-                    blockUser(eq(USER_ID))
-                }
+                coVerify { arrangement.blockUser(eq(USER_ID)) }
                 assertEquals(InfoMessageType.BlockingUserOperationError.uiText, awaitItem())
-                assertEquals(null, otherUserProfileScreenViewModel.blockUserDialogState)
+                assertEquals(null, viewModel.blockUserDialogState)
             }
         }
 
     @Test
-    fun `given a userId, when blocking user is succeed, then returns show Success message and dismiss BlockDialog`() =
+    fun `given a userId, when blocking user is succeed, then show Success message and dismiss BlockDialog`() =
         runTest {
             // given
-            coEvery { blockUser(any()) } returns BlockUserResult.Success
-            initViewModel()
-            otherUserProfileScreenViewModel.infoMessage.test {
+            val (arrangement, viewModel) = OtherUserProfileViewModelArrangement()
+                .withBlockUserResult(BlockUserResult.Success)
+                .arrange()
+            viewModel.infoMessage.test {
                 val userName = "some name"
 
                 // when
                 expectNoEvents()
-                otherUserProfileScreenViewModel.onBlockUser(USER_ID, userName)
+                viewModel.onBlockUser(USER_ID, userName)
 
                 // then
-                coVerify {
-                    blockUser(eq(USER_ID))
-                }
+                coVerify { arrangement.blockUser(eq(USER_ID)) }
                 assertEquals(
                     (awaitItem() as UIText.StringResource).resId,
                     (InfoMessageType.BlockingUserOperationSuccess(userName).uiText as UIText.StringResource).resId
                 )
-                assertEquals(null, otherUserProfileScreenViewModel.blockUserDialogState)
+                assertEquals(null, viewModel.blockUserDialogState)
             }
         }
 
-    // todo: add tests for cancel request
+    @Test
+    fun `given not connected user, then direct conversation is not requested`() = runTest {
+        //giver
```

Comment:
I think detect will throw error if there are no empty space between slashes and text